### DISCUSSION TITLE
Changes to ecs-deploy runner module not working on plan

### DISCUSSION THREAD

**Discussion by sewmiuraj** [gruntwork-io/knowledge-base#637] [How do I] [answered]
Hello,

I am facing two different issues in the ecs-deploy-runner module at: "git::git@github.com:gruntwork-io/terraform-aws-service-catalog.git//modules/mgmt/ecs-deploy-runner" using ref=v0.67.0.

ISSUE 01: 
1. Upgrading from v.0.67.0 tov 0.95.1

On the plan I am receiving the following error. Is there anything I need to be doing from my end to be able to switch to the version 0.95.1.

![Screenshot 2023-01-17 at 3 58 32 PM](https://user-images.githubusercontent.com/96437456/212932498-fc275c32-3305-4aad-b8e7-ed24e5925c19.png)

ISSUE 02:
I am trying to change the cloudwatch log group retention in days for ecs-deploy-runner. 

 But, when I set 

```
inputs = {

  # Setting up cloudwatch log group retention (in days)
  invoker_lambda_cloudwatch_log_group_retention_in_days = 2

}
```

in `terragrunt.hcl `for ecs-deploy runner - the plan returns with **No changes. Your infrastructure matches the configuration**.. But I can confirm that the cloudwatch log group's retention in the aws console is set to 3, and should therefore not return as **No changes**.
If I make changes to other variables such as `container_max_memory` it would be reflected correctly on the plan.

---

<ins datetime="2023-01-17T14:58:55Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109799">Tracked in ticket #109799</a></p>
</ins>

**Comment by josh-padnick** [comment]
Hi @sewmiuraj. I'll check with a Gruntwork Pipelines expert here and get back to you shortly!

**Comment by ellisonc** [comment, marked as answer]
@sewmiuraj for issue number 1, this is caused by a limitation of terraform.  It requires a provider to be provisioned for each AWS region. The fix is to update `multi_region_common.hcl` to have the latest list of regions.  Here is a snippet of what the current list of regions should look like:

```
 all_aws_regions = [
    "af-south-1",
    "ap-east-1",
    "ap-northeast-1",
    "ap-northeast-2",
    "ap-northeast-3",
    "ap-south-1",
    "ap-south-2",
    "ap-southeast-1",
    "ap-southeast-2",
    "ap-southeast-3",
    "ca-central-1",
    "cn-north-1",
    "cn-northwest-1",
    "eu-central-1",
    "eu-central-2",
    "eu-north-1",
    "eu-south-1",
    "eu-south-2",
    "eu-west-1",
    "eu-west-2",
    "eu-west-3",
    "me-south-1",
    "me-central-1",
    "sa-east-1",
    "us-east-1",
    "us-east-2",
    "us-gov-east-1",
    "us-gov-west-1",
    "us-west-1",
    "us-west-2"
  ]
```

Your second issue is slightly more involved, I'll do some investigation and get back to you.

**Reply by josh-padnick** [reply]
For the second issue, this [unrelated GitHub issue](https://github.com/hashicorp/terraform/issues/6169) shows an example of a state problem with this variable. I wonder if there's a similar issue here?

**Reply by sewmiuraj** [reply]
Thank you, @ellisonc  and @josh-padnick .  After fixing issue number 1 with the recommended solution, I was able to upgrade the version to v0.95.1. Afterwards, issue 2 was automatically fixed.
